cmd/istio-cni/kubernetes.go

```diff
@@ -91,3 +92,17 @@ func getK8sPodInfo(client *kubernetes.Clientset, podName, podNamespace string) (
 
 	return containers, pod.Labels, pod.Annotations, ports, nil
 }
+
+// getK8sInitContainers returns init containers of a POD
+func getK8sPodInitContainers(client *kubernetes.Clientset, podName, podNamespace string)(
+	initContainers map[string]struct{}, err error) {
+	pod, err := client.CoreV1().Pods(podNamespace).Get(podName, metav1.GetOptions{})
```

Comment:
Avoid doing another k8s API call.
Instead, modify the function above to return the info we need.

Reply:
Done